### PR TITLE
[New Package] helix editor

### PR DESCRIPTION
**Pull request by MrAdityaAlok**
Helix is s post-modern modal text editor written in rust.

### CODE REVIEW

File: packages/helix/build.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+TERMUX_PKG_DESCRIPTION="A post-modern modal text editor written in rust"
+
+# License.
+TERMUX_PKG_LICENSE="MPL-2.0"
+
+# Who cares about package.
+# Specify yourself (Github nick, or name + email) if you wish to maintain the
+# package, fix its bugs, etc. Otherwise specify "@termux".
+# Please note that unofficial repositories are not allowed to reference @termux
+# as their maintainer.
```
Comment (xeffyr) [inline packages/helix/build.sh:14]: Remove the comments.

File: packages/helix/build.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#TERMUX_PKG_SRCURL=https://github.com/helix-editor/helix/archive/v${TERMUX_PKG_VERSION}.tar.gz
+
+# SHA-256 checksum of the source code archive.
+#TERMUX_PKG_SHA256=ba388eded1246f00b5287ad9cfdaeebf24eada42729bf61b936bb70488cfb4fd
+TERMUX_PKG_SKIP_SRC_EXTRACT=true
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_get_source(){
+  mkdir -p "$TERMUX_PKG_SRCDIR" && cd "$TERMUX_PKG_SRCDIR"
+  git clone --recurse-submodules --shallow-submodules -j8 https://github.com/helix-editor/helix --branch=v0.2.1 # this is tag, actually!
```
Comment (xeffyr) [inline packages/helix/build.sh:30]: This is inappropriate way of getting source code. Please look the ./packages directory of examples for both tarball and git methods.

File: packages/helix/build.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+termux_step_make_install(){
+  cat > "hx" << EOF
+  #!/data/data/com.termux/files/usr/bin/sh
+  HELIX_RUNTIME=${TERMUX_PREFIX}/lib/helix/runtime exec ${TERMUX_PREFIX}/lib/helix/hx/hx "\$@"
+
+EOF
+  chmod +x hx
+  mv hx ${TERMUX_PREFIX}/bin
```
Comment (xeffyr) [inline packages/helix/build.sh:40]:
```
install -Dm755 ./hx $TERMUX_PREFIX/bin/hx
```

**Review by xeffyr — CHANGES_REQUESTED** [review]
Passing CI build is nice, but build.sh formatting is not something that is allowed to be free-form.

File: packages/helix/build.sh (hunk 2 above)
Comment (xeffyr) [reply packages/helix/build.sh:30]: If default methods of getting source code that are defined in build-package.sh (./scripts/build/*) are not suitable, you still can go with your own. But please put the checkout into cache dir (`TERMUX_PKG_CACHEDIR`) and then copy it to source.

Comment (MrAdityaAlok) [reply packages/helix/build.sh:30]: Source repo neither has any branch related to specific release nor do their archive has all the submodules included. So, I can not think of any other way of getting source, but if you could suggest something.....

Comment (suhan-paradkar) [reply packages/helix/build.sh:30]: use this...
```
$TERMUX_PKG_SRCURL=https://github.com/helix-editor/helix.git
```
The script will automatically include submodules and also checkout with the correct tag if the version is right ....

Comment (MrAdityaAlok) [reply packages/helix/build.sh:30]: Thank you!

File: packages/helix/build.sh (hunk 1 above)
Comment (MrAdityaAlok) [reply packages/helix/build.sh]: Did it

File: packages/helix/build.sh (hunk 2 above)
Comment (MrAdityaAlok) [reply packages/helix/build.sh]: @xeffyr  updated

File: packages/helix/build.sh (hunk 3 above)
Comment (MrAdityaAlok) [reply packages/helix/build.sh]: Updated

**Comment by xeffyr** [comment]
@MrAdityaAlok When using tabs, heredoc end can be indented as well.

**Comment by MrAdityaAlok** [comment]
> @MrAdityaAlok When using tabs, heredoc end can be indented as well.

Okay, I did not know that.

**Review by xeffyr — APPROVED** [review]

**Pull request merged**